assets/js/modules/reader-revenue-manager/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Reader Revenue Manager module initialization.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import {
+	MODULES_READER_REVENUE_MANAGER,
+	ERROR_CODE_NON_HTTPS_SITE,
+} from './datastore/constants';
+import { SetupMain } from './components/setup';
+import { SettingsEdit, SettingsView } from './components/settings';
+import ReaderRevenueManagerIcon from '../../../svg/graphics/reader-revenue-manager.svg';
+import { isFeatureEnabled } from '../../features';
+import { isURLUsingHTTPS } from './utils/validation';
+
+export { registerStore } from './datastore';
+
+const isRrmModuleEnabled =
+	( func ) =>
+	( ...args ) => {
+		if ( isFeatureEnabled( 'rrmModule' ) ) {
+			func( ...args );
+		}
+	};
+
+export const registerModule = isRrmModuleEnabled( ( modules ) => {
+	modules.registerModule( 'reader-revenue-manager', {
+		storeName: MODULES_READER_REVENUE_MANAGER,
+		SettingsEditComponent: SettingsEdit,
+		SettingsViewComponent: SettingsView,
+		SetupComponent: SetupMain,
+		Icon: ReaderRevenueManagerIcon,
+		features: [
+			// TODO: Implement the features as part of #8845.
+		],
+		checkRequirements: async ( registry ) => {
+			const homeURL = await registry
+				.__experimentalResolveSelect( CORE_SITE )
+				.getHomeURL();
```

Comment:
`getHomeURL()` doesn't have a resolver, so this promise won't resolve. We should wait for `getSiteInfo()` to resolve and then do a regular `select()` on `getHomeURL()`:

```suggestion
			await registry
				.__experimentalResolveSelect( CORE_SITE )
				.getSiteInfo();

			const homeURL = registry.select( CORE_SITE ).getHomeURL();
```

When I apply this change and test on an HTTP site I find that I am able to click on **Set up Reader Revenue Manager** without an error occurring - it takes me to the OAuth flow and gets to the point of having activated the module.

So, this point in the QAB does not seem accurate and the QAB should be updated accordingly:

> - Clicking Set up Reader Revenue Manager will throw an error, which is expected as the setup process has yet to be implemented

Reply:
>getHomeURL() doesn't have a resolver, so this promise won't resolve. We should wait for getSiteInfo() to resolve and then do a regular select() on getHomeURL():

That makes sense 👍 

It was failing In my testing. Perhaps it was fixed with some updates when I merged `develop`. Thanks for pointing it out. I'll update the QAB.

Reply:
Updated the QAB.